CRM/Admin/Form/OptionGroup.php

```diff
@@ -90,6 +79,33 @@ public function buildQuickForm() {
     }
 
     $this->assign('id', $this->_id);
+    $this->addFormRule(['CRM_Admin_Form_OptionGroup', 'formRule'], $this);
+  }
+
+  /**
+   * Global form rule.
+   *
+   * @param array $fields
+   *   The input form values.
+   *
+   * @param $files
+   * @param $self
+   *
+   * @return bool|array
+   *   true if no errors, else array of errors
+   */
+  public static function formRule($fields, $files, $self) {
+    $errors = [];
+    if ($self->_id) {
+      $name = CRM_Core_DAO::getFieldValue('CRM_Core_DAO_OptionGrou', $self->_id, 'name');
+    }
+    else {
+      $name = CRM_Utils_String::titleToVar(strtolower($fields['title']));
+    }
+    if (!CRM_Core_DAO::objectExists($name, 'CRM_Core_DAO_OptionGroup', $self->_id)) {
+      $errors['title'] = ts('Option Group name already exists in the database. Option Group Names need to be unique');
```

Comment:
Include the name in the error message so that user knows what value is conflicting, otherwise it's not clear. Shorter message too.

$errors['title'] = ts('Option Group name: ' . $name .' already exists. Option Group Names need to be unique');